### PR TITLE
add detection and report that Manjaro isn't supported

### LINKED ISSUE #11: Manjaro has the same issue with powershell 6.0 as fedora 28, but is also not officially supported by 6.1 either
Manjaro is not officially supported by 6.1 because it is not supported by DotNet Core:
https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1-supported-os.md#linux

etc/os-release details
image

### PR DESCRIPTION
**Pull request by TravisEz13**
add detection and report that Manjaro isn't supported
  - Add workaround for the terminal issue in the stable package
  - refactor workaround into the function

fixes #11

### CODE REVIEW

**Comment by TravisEz13** [comment]
@tylerl0706  Can you review when you have time?

File: stable/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline stable/launcher/powershell-wrapper]: Something I'm concerned about is that as time goes on, and Snap supports more distros we run into more issues and have to add other workarounds like this making this file large.

Ideally, I wouldn't want a bunch of functions in this file, but pulled out to some other file if possible like a "workarounds" script.

Or a workaround script per distro with the workarounds per revision.

Since this is the only workaround so far... I wont block on this, but it's something think about for the future... unless you want to address it now :)

File: preview/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline preview/launcher/powershell-wrapper]: technically arch is community supported... would that fall under unsupported in this case?

What about the other weird distros that snap supports:
image
Solus, yocto, openwrt, etc.

**Review by TylerLeonhardt — COMMENTED** [review]
Just a couple questions before I sign off

File: preview/launcher/powershell-wrapper
Comment (TravisEz13) [reply preview/launcher/powershell-wrapper]: I need to know how to detect them.  I'll update the PR for Arch as well

**Comment by TravisEz13** [comment]
I've been thinking, we might want to just put a message on the store page pointing to our support policy and remove these messages about being unsupported.  Two reasons:

-  If we want to be accurate, we would need more a whitelist and give the message when we are not on one of those distros, but we would still need the logic for the distros with the terminal issue.
-  Updating the docs is a lot easier than updating the code.

**Comment by TravisEz13** [comment]
This is the official support matrix from DotNet for 2.1:
## Linux

OS                            | Version                       | Architectures  | Notes
------------------------------|-------------------------------|----------------|-----
Red Hat Enterprise Linux      | 6                             | x64            | [Red Hat support policy](https://access.redhat.com/support/policy/updates/errata/) 
Red Hat Enterprise Linux <br> CentOS <br> Oracle Linux | 7    | x64            | [Red Hat support policy](https://access.redhat.com/support/policy/updates/errata/) <br> [CentOS lifecycle](https://wiki.centos.org/FAQ/General#head-fe8a0be91ee3e7dea812e8694491e1dde5b75e6d) <br> [Oracle Linux lifecycle](http://www.oracle.com/us/support/library/elsp-lifetime-069338.pdf)
Fedora                        | 27, 28                    | x64            | [Fedora lifecycle](https://fedoraproject.org/wiki/End_of_life)
Debian                        | 9, 8.7+                       | x64, ARM32\*     | [Debian lifecycle](https://wiki.debian.org/DebianReleases)
Ubuntu                        | 18.04, 17.10, 16.04, 14.04    | x64, ARM32\*    | [Ubuntu lifecycle](https://wiki.ubuntu.com/Releases)
Linux Mint                    | 18, 17                        | x64            | [Linux Mint end of life announcements](https://forums.linuxmint.com/search.php?keywords=%22end+of+life%22&terms=all&author=&sc=1&sf=titleonly&sr=posts&sk=t&sd=d&st=0&ch=300&t=0&submit=Search)
openSUSE                      | 42.3+                         | x64            | [OpenSUSE lifecycle](https://en.opensuse.org/Lifetime)
SUSE Enterprise Linux (SLES)  | 12 SP2+                       | x64            | [SUSE lifecycle](https://www.suse.com/lifecycle/)
Alpine Linux                  | 3.7+                          | x64            | [Alpine lifecycle](https://wiki.alpinelinux.org/wiki/Alpine_Linux:Releases)

Meaning, rasbian, Arch, Manjaro, OpenWrt, OpenEmbedded, and Solus **cannot** be officially supported, but this is actualy a small list of where snap runs.

If you look at the distros in use at https://snapcraft.io/vscode,
You see zorin, elementary, neon are more populare that many of these community supported distros above and we also have kali, antergos, deepin and galliumos near the end of the list.

File: preview/launcher/powershell-wrapper
Comment (TylerLeonhardt) [reply preview/launcher/powershell-wrapper]: Should this be `DistroBasedOn` not `DIST`?

Comment (TylerLeonhardt) [reply preview/launcher/powershell-wrapper]: or line 55 be `DIST`

File: stable/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline stable/launcher/powershell-wrapper]: same question about `DIST` vs `DistroBasedOn`

**Comment by TylerLeonhardt** [comment]
@TravisEz13 I agree on the message on the store page. That will be the best way to have up-to-date information.

File: stable/launcher/powershell-wrapper
Comment (TravisEz13) [reply stable/launcher/powershell-wrapper]: There is already a separate script for 6.1 where DotNet has fixed this issue.  The distro detection needs to happen in one file.  For PowerShell installation, we follow the model you suggest but on the family basis.  I think we should handle it when we get big enough that it matters.  So far, this issue seems to only occur on RedHat variants (Manjaro seem to be more related to RedHat that Debian.)  If I had better detection of the issue itself, your suggestion might be useful.

File: stable/launcher/powershell-wrapper
Comment (TravisEz13) [reply stable/launcher/powershell-wrapper]: I don't have an Arch machine to get the DIST value.  The potential issue here is this section of logic might overreach, but it would be another unsupported distro, so I don't think it's an issue.

**Comment by TravisEz13** [comment]
I filed https://github.com/PowerShell/PowerShell-Snap/issues/13 to track testing of the distros that use vscode

**Comment by TravisEz13** [comment]
I filed https://github.com/PowerShell/PowerShell-Snap/issues/14 to track moving the support messages to the store

File: stable/launcher/powershell-wrapper
Comment (TravisEz13) [reply stable/launcher/powershell-wrapper]: I verified the logic works in a container:
`Warning: PowerShell 6.0 does not support arch - and only community support is available!`

File: preview/launcher/powershell-wrapper
Comment (TravisEz13) [reply preview/launcher/powershell-wrapper]: Technically another distro could base theirs on manjaro and not change the factor I use to determine DistroBasedOn.

Comment (TravisEz13) [reply preview/launcher/powershell-wrapper]: Look up on line 23 for the amazon linux vs redhat code.

**Comment by TravisEz13** [comment]
Another way of explaining it, is I only want the logic in the Manjaro and fedora sections to apply to that particular distro.  I want the logic in the arch section to apply to all other distro based on arch.

File: preview/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline preview/launcher/powershell-wrapper]: nit: "distributions" not "distrobutions"

File: stable/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline stable/launcher/powershell-wrapper]: nit: "distributions" not "distrobutions"

File: stable/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline stable/launcher/powershell-wrapper]: nit: "distros" or "distributions" to be consistent

File: stable/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline stable/launcher/powershell-wrapper]: optional nit... change to "distribution" to be consistent.

File: stable/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline stable/launcher/powershell-wrapper:89]: nit: "distributions" not "distrobutions"

File: preview/launcher/powershell-wrapper
Comment (TylerLeonhardt) [inline preview/launcher/powershell-wrapper:78]: "distributions" not "distrobutions"

**Review by TylerLeonhardt — APPROVED** [review]
All just a single nitpick... "distrobutions" to "distributions"

otherwise, LGTM!

**Pull request merged**
